DoSomething/rogue
# Adds Quantity and Why Participated to Uploader

#### What's this PR do?
- Adds optional quantity and why participated fields in the uploader for admins to populate a user's first post or also change these fields when uploading a post on their behalf. 
- Adds language that warns admins that the post will automatically be approved, that they'll have to add tags in Rogue, and that the user will not receive an email if they upload a photo. These don't look like the wireframes in the [Pivotal Card](https://www.pivotaltracker.com/n/projects/2019429/stories/151067785) but figured we can revisit this when @lkpttn @ngjo are back since the uploader is a little different from these anyway! 

#### How should this be reviewed?
- Go to a user's signup page (`/signups/:signup_id`). 
- Click on "Upload Photo" and you should see the below: 
![screen shot 2017-10-24 at 2 33 56 pm](https://user-images.githubusercontent.com/9019452/31961329-67acca8e-b8c8-11e7-8203-5ddb59d31b39.png)
- Try to upload a photo with caption and it should go through. 
- Try to upload another photo with caption, why, and quantity and also should go through and update on the signup page (and database). 

#### Any background context you want to provide?
- A few changes from the [Pivotal Card](https://www.pivotaltracker.com/n/projects/2019429/stories/151067785):
  - The last task says "Adding in text to explain that quantity + why are optional if they've provided it before - @luke what's the best language for this!?" Instead of adding this text since @ngjo and @lkpttn are out this week, I added language the placeholders saying that these are optional. We can change these later according to what @lkpttn thinks is best! 
  - Same with the first task - not exactly like the wireframe but we can change these as well later (more info. above).  
  - The [Pivotal Card](https://www.pivotaltracker.com/n/projects/2019429/stories/151067785) also mentions that sometimes, a user will ask admins/campaign leads to fix their "why participated" statement. This PR doesn't allow you to do this - you _must_ upload a photo in order to update the "quantity" or "why participated" in this modal. I think updating the "why" belongs in the "Edit | Show History" modal similar (since you can also update the "quantity" here without adding a new photo). Here's a [placeholder card](https://www.pivotaltracker.com/n/projects/2019429/stories/152252363) to add this and discuss when @ngjo and @lkpttn are back! 

#### Relevant tickets
Fixes this [Pivotal Card](https://www.pivotaltracker.com/n/projects/2019429/stories/151067785).

#### Checklist
